### PR TITLE
Add args for async SGD

### LINKED ISSUE #1136: Add arguments for async-SGD and pass them to master/worker
According to [design doc](https://github.com/sql-machine-learning/elasticdl/blob/develop/elasticdl/doc/async_sgd_design.md#add-arguments-for-elasticdltrain), add 3 arguments:

- --use_async, default=False, help="True for asynchronous SGD, False for synchronous SGD"
- --lr_staleness_modulation, default=False, help="If True, master will modulate learning rate with staleness in asynchronous SGD"
- --get_model_frequency, default=1, help="worker will get_model from PS every this steps."

Pass `use_async` and `lr_staleness_modulation` to master.
Pass `use_async` and `get_model_frequency` to worker.

### PR DESCRIPTION
**Pull request by chunyang-wen**
closes #1136

### CODE REVIEW

File: elasticdl/python/common/args.py [hunk 1]
```diff
@@ -268,6 +268,23 @@ def add_train_params(parser):
...
     )
+    parser.add_argument(
+        "--use_async",
+        default=False,
+        help="True for asynchronous SGD, False for synchronous SGD",
+    )
+    parser.add_argument(
+        "--lr_staleness_modulation",
+        default=False,
+        help="If True, master will the modulate learning rate with staleness "
```
Comment (skydoorkai) [inline elasticdl/python/common/args.py]: master will modulate the learning rate with staleness

File: elasticdl/python/common/args.py [hunk 2]
```diff
@@ -268,6 +268,23 @@ def add_train_params(parser):
...
+        "in asynchronous SGD",
+    )
+    parser.add_argument(
+        "--get_model_frequency",
+        type=int,
+        default=1,
+        help="worker will get_model from PS every this steps.",
```
Comment (skydoorkai) [inline elasticdl/python/common/args.py]: every these steps

File: elasticdl/python/worker/worker.py [hunk 3]
```diff
@@ -39,6 +39,7 @@ def __init__(
         model_params="",
         prediction_outputs_processor="PredictionOutputsProcessor",
         max_minibatch_retry_num=DEFAULT_MAX_MINIBATCH_RETRY_NUM,
+        get_model_frequency=1,
```
Comment (skydoorkai) [inline elasticdl/python/worker/worker.py]: Also need to assign `get_model_frequency` in __init__ to self.

**Review by skydoorkai — COMMENTED** [review]
Need to pass the argument `use_async`  to servicer.

File: elasticdl/python/worker/worker.py (hunk 3 above)
Comment (chunyang-wen) [reply elasticdl/python/worker/worker.py]: I planed to do this in upcoming commits.

File: elasticdl/python/worker/main.py [hunk 4]
```diff
@@ -80,6 +80,15 @@ def _parse_args():
         "e.g. \"{'ip_0': [port_0,port_1]}\"",
     )
 
+    # TODO (chengfu.wcy) Refact arguments of master and worker
+    # to avoid duplicate decalaration.
+    parser.add_argument(
+        "--get_model_frequency",
+        type=int,
+        default=1,
+        help="worker will get_model from PS every this steps.",
```
Comment (terrytangyuan) [inline elasticdl/python/worker/main.py]: worker -> Worker
this steps -> this many steps

File: elasticdl/python/common/args.py [hunk 5]
```diff
@@ -268,6 +268,23 @@ def add_train_params(parser):
...
+        help="True for asynchronous SGD, False for synchronous SGD",
+    )
+    parser.add_argument(
+        "--lr_staleness_modulation",
+        default=False,
+        help="If True, master will modulate the learning rate with staleness "
+        "in asynchronous SGD",
+    )
+    parser.add_argument(
+        "--get_model_frequency",
```
Comment (terrytangyuan) [inline elasticdl/python/common/args.py]: This is weird to be an argument name, maybe something like "model_refresh_steps"? The use of "steps" is to be consistent with other args.

Comment (chunyang-wen) [reply elasticdl/python/common/args.py]: How about `model_stale_steps`?
`sync_model_steps` ?

Comment (skydoorkai) [reply elasticdl/python/common/args.py]: the local model is updated every minibatch. This is the number of steps for workers to get_model from the master.

**Review by skydoorkai — APPROVED** [review]

**Pull request merged**